tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -658,3 +660,23 @@ def test_params_to_tune(reconciliator, model, transforms, expected_params_to_tun
     obtained_params_to_tune = pipeline.params_to_tune()
 
     assert obtained_params_to_tune == expected_params_to_tune
+
+
+@pytest.mark.parametrize(
+    "ts_name",
+    (
+        "market_level_constant_forecast_with_target_components",
+        "market_level_constant_forecast_with_quantiles",
+        "market_level_constant_hierarchical_ts_w_exog",
+        "market_level_constant_hierarchical_ts",
+    ),
+)
+def test_make_hierarchical_dataset(ts_name, request):
+    ts = request.getfixturevalue(ts_name)
+
+    new_ts = HierarchicalPipeline._make_hierarchical_dataset(ts, hierarchical_structure=ts.hierarchical_structure)
```

Comment:
It looks like this method could just make a deep copy of `ts` and pass the test.